types/serverless/index.d.ts

```diff
@@ -0,0 +1,102 @@
+// Type definitions for serverless 1.18
+// Project: https://github.com/serverless/serverless#readme
+// Definitions by: Hassan Khan <https://github.com/hassankhan>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare namespace Serverless {
+    interface Options {
+        stage: string | null;
+        region: string | null;
+        noDeploy?: boolean;
+    }
+
+    namespace Provider {
+        class Aws {
+            constructor(serverless: Serverless, options: Options)
+
+            getProviderName(): string;
+            getRegion(): string;
+            getServerlessDeploymentBucketName(): string;
+            getStage(): string;
+        }
+    }
+
+    namespace Service {
+        interface Custom {
+            [key: string]: any;
+        }
+    }
+
+    interface Config {
+        servicePath: string;
+    }
+
+    class YamlParser {
+        constructor(serverless: Serverless)
+        parse(yamlFilePath: string): Promise<any>;
+    }
+
+    interface FunctionDefinition {
+        name: string;
+    }
+
+    interface Event {
+        eventName: string;
+    }
+
+    class Service {
```

Comment:
Instead of merging this in this file, may be add file classes/service.d.ts to `export = Service` to handle this more appropriately. (same for all the other types esp classes you are adding here to the namespace.

Reply:
Sorry for the noob question, but if I move the class declaration to another file, how would I keep it under the Serverless namespace?

Reply:
Don't think they should be under Serverless namespace especially since they are in different module.

Reply:
Add a file called `classes/service.d.ts` with `class Service` declaration. Add `export = Service` as well. In index.d.ts you do
`import Service = require("./classes/service")`

Reply:
Thanks 😄